lib/CoreReplicator.js

```diff
@@ -0,0 +1,46 @@
+/**
+* @module CoreReplicator
+ */
+
+import { CoreIdCache } from "./authstore/coreIdCache"
+import { Sqlite } from "./sqlite"
+
+export class CoreReplicator {
+  /** @type {string} */
+  #coreId
+  #masterCore
+  /** @type {Corestore} */
+  #store
+  /** @type {CoreIdCache} */
+  #coreIdCache
+  /** @type {string} */
+  #extensionNamespace = 'replicator'
+  /** @type {any} */
+  #extension
+  /**
+* Create a CoreReplicator instance. This class is in charge of replicating cores belonging to a namespace in a corestore with other peers. This is acomplished by registering an hypercore extension.
+* @param {string} coreId - core public key to use as master core to register the extension in
+* @param {Corestore} store - set of cores to share
+* @param {CoreIdCache} coreIdCache  
+  */
+  constructor(coreId, store, coreIdCache){
```

Comment:
It might be best to pass in the core rather than the coreId. This is something that I'm changing in DataType code right now, too, so that it takes a core rather than a keyPair for a core.